tests/app/views/contexts/test_calculated_summary_context.py

```diff
@@ -9,237 +7,115 @@
 from app.questionnaire.questionnaire_schema import DEFAULT_LANGUAGE_CODE
 from app.questionnaire.routing_path import RoutingPath
 from app.utilities.schema import load_schema_from_name
-from app.views.contexts import SectionSummaryContext
 from app.views.contexts.calculated_summary_context import CalculatedSummaryContext
-from tests.app.views.contexts import SummaryContextTestCase
-
-
-class TestCalculatedSummaryContext(SummaryContextTestCase):
-    def setUp(self):
-        super().setUp()
-        self.schema = load_schema_from_name("test_calculated_summary")
-        answers = [
-            {"value": 1, "answer_id": "first-number-answer"},
-            {"value": 2, "answer_id": "second-number-answer"},
-            {"value": 3, "answer_id": "second-number-answer-unit-total"},
-            {"value": 4, "answer_id": "second-number-answer-also-in-total"},
-            {"value": 5, "answer_id": "third-number-answer"},
-            {"value": 6, "answer_id": "third-and-a-half-number-answer-unit-total"},
-            {"value": "No", "answer_id": "skip-fourth-block-answer"},
-            {"value": 7, "answer_id": "fourth-number-answer"},
-            {"value": 8, "answer_id": "fourth-and-a-half-number-answer-also-in-total"},
-            {"value": 9, "answer_id": "fifth-percent-answer"},
-            {"value": 10, "answer_id": "fifth-number-answer"},
-            {"value": 11, "answer_id": "sixth-percent-answer"},
-            {"value": 12, "answer_id": "sixth-number-answer"},
-        ]
-        self.block_type = "CalculatedSummary"
-        self.language = "en"
-        self.metadata = {}
-        self.response_metadata = {}
-        self.answer_store = AnswerStore(answers)
-        self.list_store = ListStore()
-        self.progress_store = ProgressStore()
-
-    @patch("app.jinja_filters.flask_babel.get_locale", Mock(return_value="en_GB"))
-    def test_build_view_context_for_currency_calculated_summary_no_skip(self):
-        current_location = Location(
-            section_id="default-section", block_id="currency-total-playback-with-fourth"
-        )
-
-        calculated_summary_context = CalculatedSummaryContext(
+from app.views.contexts.section_summary_context import SectionSummaryContext
+from tests.app.views.contexts import assert_summary_context
+
+
+# pylint: disable=too-many-locals
+@pytest.mark.usefixtures("app")
+@pytest.mark.parametrize(
+    "block_id, locale, language, title, value, block_len, answers",
```

Comment:
`block_len` sounds like the length of the block than the number of blocks you expect. Might be clearer as `total_blocks` or `number_of_blocks`